### ISSUE TITLE
make ndtest(axes) use those axes and maybe deprecate ndrange

### ISSUE THREAD

**Issue by gdementen** [liam2/larray#534]
Currently, it only keeps the size of the axis, which is mostly useless. And since ndrange is only used for testing (now that sequence exists), it would make sense to only have (or at least use) one function.

The current behaviour is:

```python
>>> ndrange(3)
{0}*  0  1  2
      0  1  2
>>> ndrange('c=C2..C4')
c  C2  C3  C4
    0   1   2
>>> ndtest(3)
a  a0  a1  a2
    0   1   2
>>> ndtest('c=C2..C4')
a  a0  a1  a2
    0   1   2
```

While I think it should be:

```python
>>> ndtest(3)
a  a0  a1  a2
    0   1   2
>>> ndtest('c=C2..C4')
c  C2  C3  C4
    0   1   2
>>> ndtest('0..2')
{0}  0  1  2
     0  1  2
```
The only problem if we deprecate ndrange is that generating ranges using wildcard axes is a bit more tedious (though the very few examples where we currently use that in tests would actually be better if we used ndtest instead).

```python
>>> ndtest((Axis(2), Axis(3)))
{0}*\{1}*  0  1  2
        0  0  1  2
        1  3  4  5
```

We could also implement something like this:

```python
>>> ndtest((2, 3), wildcard_axes=True)
{0}*\{1}*  0  1  2
        0  0  1  2
        1  3  4  5
```

In short, I am unsure about deprecating ndrange but I am pretty sure we should only use ndtest in doctests.

**Comment by alixdamman** [comment]
Simplier API (less methods/functions) is often better (at least to maintain). 
So I'm OK to extend behavior of `ndtest` and also to deprecate `ndrange` (except us who is using `ndrange` ???).

Concerning the implementation, let's start with:

```python
>>> ndtest(3)
a  a0  a1  a2
    0   1   2
>>> ndtest('c=C2..C4')
c  C2  C3  C4
    0   1   2
>>> ndtest('0..2')
{0}  0  1  2
     0  1  2
```

And later or if I have enough time to add it: 

```python
>>> ndtest((2, 3), wildcard_axes=True)
{0}*\{1}*  0  1  2
        0  0  1  2
        1  3  4  5
```

**Comment by alixdamman** [comment]
reminder : update api.rst + getting_started.rst + tutorial.rst
